### PR TITLE
fix(DataPicker):fix picker content  is not centered

### LINKED ISSUE #44235: DataPicker 组件的 table 未居中
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/rr2g68)

### Steps to reproduce

检查选择器的日期 table

### What is expected?

日期 table 居中

### What is actually happening?

日期 table 宽度比预期宽。
多余量 = `列数 * 2px`
image

| Environment | Info |
| --- | --- |
| antd | 5.8.3 |
| React | 18.2.0 |
| System | macOS Ventura 13.4.1 (c) |
| Browser | Chrome Version 114.0.5735.198 (Official Build) (arm64) |

---
原因是日期的 th 包含 `padding: 1px`，且对浏览器的默认样式未进行 reset 的情况下，th 的 `box-sizing` 会是 `content-box`，导致计算出的 th 的宽度并不包含两侧的 padding。日期 table 的宽度便会多出 `列数 * 2px`。

### PR DESCRIPTION
**Pull request by Zian502**
fix(DataPicker):fix picker content  is not centered

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/44235

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix(DataPicker):fix picker content is not centered      |
| 🇨🇳 Chinese |      修复日期组件的内容不居中问题     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at d0dbef6</samp>

Fix date picker panel overflow bug by adding `boxSizing` property to panel style in `components/date-picker/style/index.ts`.

### <samp>🤖 Generated by Copilot at d0dbef6</samp>

* Fix date picker panel overflow bug by adding `boxSizing` property to panel style ([link](https://github.com/ant-design/ant-design/pull/44245/files?diff=unified&w=0#diff-a4f33cb3f0dfef43aa860e509b2c43fc62168efd797b258ed2cf47dfc4b1eaa4R810))

### CODE REVIEW

File: components/date-picker/style/index.ts [hunk 1]
```diff
@@ -807,6 +807,7 @@ export const genPanelStyle = (token: SharedPickerToken): CSSObject => {
 
           th: {
             width: pickerPanelCellWidth,
+            boxSizing:'border-box',
```
Comment (afc163) [inline components/date-picker/style/index.ts:810]:
```suggestion
            boxSizing: 'border-box',
```

**Comment by afc163** [comment]
🔗 Related issue link 

这里留错链接了。

File: components/date-picker/style/index.ts (hunk 1 above)
Comment (afc163) [reply components/date-picker/style/index.ts:810]: @Zian502

Comment (Zian502) [reply components/date-picker/style/index.ts:810]: > 看 issue 讨论添加 `padding: 0` 更好吧

根本原因没有 reset th 标签默认样式，默认会有个 padding 值，改成 padding: 0 代码理解会更好些，其它的我还没有想到更好的点
image

Comment (Zian502) [reply components/date-picker/style/index.ts:810]: 有个疑问，reset.css 这里可以加 th { padding: 0 } 嘛？
image

Comment (MadCcc) [reply components/date-picker/style/index.ts:810]: `reset.css` 自带了 `padding: 0`，这里要解的是没有引入 `reset.css` 时的问题，所以直接在 DatePicker 的样式里给 th 添加 `padding: 0` 就行。

Comment (MadCcc) [reply components/date-picker/style/index.ts:810]: 应该考虑没有 `reset.css` 的情况。`box-sizing` 和 `padding` 都加上吧，这样稳定些

Comment (Zian502) [reply components/date-picker/style/index.ts]: > 应该考虑没有 `reset.css` 的情况。`box-sizing` 和 `padding` 都加上吧，这样稳定些

done.

**Review by afc163 — APPROVED** [review]

**Pull request merged**